src/main/java/com/couchbase/lite/replicator/Replication.java

```diff
@@ -406,20 +404,76 @@ public int getChangesCount() {
     /**
      * Update the lastError
      */
-    /* package */ void setLastError(Throwable lastError) {
+    protected void setLastError(Throwable lastError) {
         this.lastError = lastError;
     }
 
-    /* package */ String remoteCheckpointDocID() {
+    /**
+     * Following two methods for temporary methods instead of CBL_ReplicatorSettings implementation.
+     */
+    protected String remoteCheckpointDocID() {
         return replicationInternal.remoteCheckpointDocID();
     }
 
+    public Set<String> getPendingDocumentIDs(){
+        if(isPull() || (isRunning() && pendingDocIDs != null))
+            return pendingDocIDs;
+
+        final CountDownLatch latch = new CountDownLatch(1);
+        pendingDocIDs = new HashSet<String>();
+        this.workExecutor.submit(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    String filterName = (String) properties.get("filter");
+                    Map<String, Object> filterParams = (Map<String, Object>) properties.get("query_params");
+                    ReplicationFilter filter = null;
+                    if (filterName != null) {
+                        filter = db.getFilter(filterName);
+                    }
+
+                    String lastSequence = replicationInternal.lastSequence;
+                    if (lastSequence == null)
+                        lastSequence = db.lastSequenceWithCheckpointId(remoteCheckpointDocID());
+                    RevisionList revs = db.unpushedRevisionsSince(lastSequence, filter, filterParams);
+                    Log.e(Log.TAG_SYNC, revs.toString());
```

Comment:
I saw the line below checking `revs != null`. If revs is null, this will be NPE.